plugins/woocommerce/src/Blocks/Domain/Bootstrap.php

```diff
@@ -122,6 +124,35 @@ function () {
 			0
 		);
 
+		// We need to initialize BlockTemplatesController and BlockTemplatesRegistry at the end of `after_setup_theme`
+		// so themes had the opportunity to declare support for template parts.
+		add_action(
+			'after_setup_theme',
+			function () {
+				$is_rest = wc()->is_rest_api_request();
+				// phpcs:disable WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+				$is_store_api_request = $is_rest && ! empty( $_SERVER['REQUEST_URI'] ) && ( false !== strpos( $_SERVER['REQUEST_URI'], trailingslashit( rest_get_url_prefix() ) . 'wc/store/' ) );
```

Comment:
Shall we turn this into its own method inside this class since we're not using it twice in `Bootstrap.php`?

Reply:
There is actually already a method on the `WooCommerce` class [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-woocommerce.php#L477-L488). Is there any reason we can't/aren't using this?

Reply:
Oh, good catch! I can't think of any reason for not using the `WooCommerce` method. Should be solved in 1a5ca98e7d83d3d4ba07f6d4a8e675b2837dd119.